tiflash/tiflash-disaggregated-and-s3.md

```diff
@@ -0,0 +1,166 @@
+
+  Compute Node 是无状态节点，它拥有秒级的扩容和缩容速度。你可以利用这个特性降低成本：
+  
+  - 在查询负载较低时，减少 Compute Node 的数量，从而节省成本。在没有查询时，甚至可以停掉所有 Compute Node。
+  - 在查询负载变高时，快速增加 Compute Node 的数量，保证查询性能。
+
+## 使用场景
+
+TiFlash 存算分离架构适合于希望获得更高性价比的数据分析服务的场景。在这个架构下，存储和计算资源可以单独按需扩展。在这些场景将会有较大收益：
+
+- 数据量虽然很大，但是只有少量数据被频繁查询；其他大部分数据属于冷数据，很少被查询。此时经常被查询的数据通常已被缓存在 Compute Node 的本地 SSD 上，可以提供较快查询性能；而其他大部分冷数据则存储在成本较低的 S3 或者其他对象存储上，从而节省存储成本。
+
+- 计算资源需求有明显的波峰和波谷。比如重型的对账查询通常放在晚上执行，此时对计算资源要求较高，可以考虑临时扩展 Compute Node；其他时间可以用较少的 Compute Node 完成查询任务。
+
+## 准备条件
+
+准备一个用于存储 TiFlash 数据的 S3 bucket。可以使用已有的 bucket，但是每个 TiDB 集群需要有专门的存储目录。
+
+## 使用方式
+
+默认情况下，TiUP 会将 TiFlash 部署为存算一体架构。如需将 TiFlash 部署为存算分离架构，请参考以下步骤手动进行配置：
+
+1. 确保 TiDB 集群中没有任何 TiFlash 节点。如果有，则需要将所有表的 TiFlash 副本数设置为 0，然后缩容掉所有 TiFlash 节点。比如：
+
+  ```sql
+  SELECT * FROM INFORMATION_SCHEMA.TIFLASH_REPLICA; # 查询所有带有 TiFlash 副本的表
+  ALTER TABLE table_name SET TIFLASH REPLICA 0;     # 将所有表的 TiFlash 副本数设置为 0
+  ```
+
+  ```shell
+  tiup cluster scale-in mycuster -R tiflash # 缩容掉所有 TiFlash 节点
+  tiup cluster display mycluster            # 等待所有 TiFlash 节点进入 Tombstone 状态
+  tiup cluster prune mycluster              # 移除所有处于 Tombstone 状态的 TiFlash 节点
+  ```
+
+2. 准备 TiFlash 的拓扑配置文件，比如 scale-out.topo.yaml，配置内容如下：
+
+  ```yaml
+  tiflash_servers:
+    # TiFlash 节点存在 storage.s3 配置说明使用存算分离模式
+    # 如果配置了 flash.disaggregated_mode: tiflash_compute，则节点类型是 Compute Node；否则是 Write Node
+
+    # 172.31.8.1~2 是 TiFlash Write Node
+    - host: 172.31.8.1
+      config:
+        flash.disaggregated_mode: tiflash_write               # 这是一个 Write Node
+        storage.s3.endpoint: http://s3.{region}.amazonaws.com # S3 的 endpoint 地址
+        storage.s3.bucket: my_bucket                          # TiFlash 的所有数据存储在这个 bucket 中
+        storage.s3.root: /cluster1_data                       # S3 bucket 中存储数据的根目录
+        storage.s3.access_key_id: {ACCESS_KEY_ID}             # 访问 S3 的 ACCESS_KEY_ID
+        storage.s3.secret_access_key: {SECRET_ACCESS_KEY}     # 访问 S3 的 SECRET_ACCESS_KEY
+        storage.main.dir: ["/data1/tiflash/data"]             # Write Node 的本地数据目录，和存算一体的配置方式相同
+    - host: 172.31.8.2
+      config:
+        flash.disaggregated_mode: tiflash_write               # 这是一个 Write Node
+        storage.s3.endpoint: http://s3.{region}.amazonaws.com # S3 的 endpoint 地址
+        storage.s3.bucket: my_bucket                          # TiFlash 的所有数据存储在这个 bucket 中
+        storage.s3.root: /cluster1_data                       # S3 bucket 中存储数据的根目录
+        storage.s3.access_key_id: {ACCESS_KEY_ID}             # 访问 S3 的 ACCESS_KEY_ID
+        storage.s3.secret_access_key: {SECRET_ACCESS_KEY}     # 访问 S3 的 SECRET_ACCESS_KEY
+        storage.main.dir: ["/data1/tiflash/data"]             # Write Node 的本地数据目录，和存算一体的配置方式相同
+
+    # 172.31.9.1~2 是 TiFlash Compute Node
+    - host: 172.31.9.1
+      config:
+        flash.disaggregated_mode: tiflash_compute             # 这是一个 Compute Node
```

Comment:
```suggestion
      config:
        delta_index_cache_size: 2000
        flash.disaggregated_mode: tiflash_compute             # 这是一个 Compute Node
```

Reply:
Let's don't put `delta_index_cache_size` in user document. Because we are very likely to change it in later version.